warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -741,6 +609,183 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
         return edgesCreated;
     }
     
+    private boolean shortCircuit(RawRecordContainer event, String typeName) {
+        if (event.fatalError()) {
+            return true;
+        }
+        if (!edges.containsKey(typeName)) {
+            return true;
+        }
+        return false;
+    }
+    
+    private boolean eventLacksEdgeFields(EdgeDefinition edgeDef) {
+        
+        Multimap<String,NormalizedContentInterface> mSource = null;
+        Multimap<String,NormalizedContentInterface> mSink = null;
+        
+        if (depthFirstList.containsKey(edgeDef.getSourceFieldName()) && depthFirstList.containsKey(edgeDef.getSinkFieldName())) {
+            mSource = depthFirstList.get(edgeDef.getSourceFieldName());
+            mSink = depthFirstList.get(edgeDef.getSinkFieldName());
+        } else {
+            return true;
+        }
+        
+        // bail if the event doesn't contain any values for the source or sink field
+        if (null == mSource || null == mSink) {
+            return true;
+        }
+        if (mSource.isEmpty() || mSink.isEmpty()) {
+            return true;
+        }
+        
+        return false;
+    }
+    
+    private long getActivityDate() {
+        if (normalizedFields.containsKey(edgeDefConfigs.getActivityDateField())) {
+            String actDate = normalizedFields.get(edgeDefConfigs.getActivityDateField()).iterator().next().getEventFieldValue();
+            try {
+                return DateNormalizer.parseDate(actDate, DateNormalizer.FORMAT_STRINGS).getTime();
+            } catch (ParseException e1) {
+                log.error("Parse exception when getting the activity date: " + actDate + " for edge creation " + e1.getMessage());
+            }
+        }
+        return 0L;
+    }
+    
+    private String getEdgeAttr3(EdgeDefinitionConfigurationHelper edgeDefConfigs) {
```

Comment:
Just to reiterate, I support what you're doing here.  It's a step in the right direction.

This datatype handler is doing a lot.  I wonder if some of the newly extracted methods could be gathered into new classes to help with encapsulation and readability.  

Here's a (partial) list of concerns for the DataTypeHandler that might help guide the design of new classes:
- Extract event characteristics, e.g. dates, to be used for edges
- Generate a single EdgeDataBundle for a given event and edge definition
- Generate all the EdgeDataBundle objects for an event, according to its edge definitions 
- Functions that operate on an EdgeDataBundle, like deny list application, masking, etc.
- Create Accumulo Key(s) for EdgeDataBundle
 
Ideas for classes based on the above concerns:
1. One such class might be responsible for generating an EdgeDataBundle object for a particular EdgeDefinition.  Maybe the builder pattern would be appropriate here.

2. Another class could take a new event, extract just once its activity date, determine its date type, etc., and then loop through the edge definitions, calling the builder (item 1) for each edge definition until they are all created.

3. writeEdges could be the basis for another class that takes an EdgeDataBundle and a context writer.  It would then write the appropriate keys to the writer.  It's already using an EdgeKeyBuilder, so that part is already abstracted out.

4. The mask and deny list capabilities seem to suggest creating an interface that operates on an EdgeDataBundle and an event.  Maybe item 2 could call these interfaces.

I think the approach outlined here could address some of the comments like the one about reparsing activity date.  

A lot of the methods have already been extracted through the existing refactor, so much of this would be grouping them together into new classes.  One goal for the proposed new classes would be to have unit tests for each with minimal dependencies / integration points.  Maybe a lot of this is already part of the plan as I see it's a WIP.  Let me know if you want a hand, even if it's just adding tests or something.

Reply:
Cool ideas.  I'd rather implement them in phases, which is why I was starting with breaking blocks up into their own methods first.

Reply:
I'm just going to go ahead and implement these anyway.  They're crawling out of the woodwork so naturally.... EdgeIngestConfiguration, EdgeEventFieldInfo, etc forthcoming

Reply:
I've broken out the classes and am pretty happy with where it landed.  There is probably more we could do, but I think this moves the ball forward significantly